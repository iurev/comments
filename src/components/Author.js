import React from 'react'
import styled from 'styled-components'
import { UserCircle } from 'styled-icons/boxicons-solid/UserCircle'


const Root = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: 10px;
`

const AvatarWrapper = styled.div`
  width: 40px;
  height: 40px;
  border-radius: 20px;
  margin-right: 10px;
  overflow: hidden;
`

const Avatar = styled.img`
  width: 100%;
  height: 100%;
  transform: scale(1);
  transition: transform ease-in-out 0.2s;

  &:hover {
    transform: scale(1.2);
  }
`

const DefaultAvatar = styled(UserCircle)`
  width: 100%;
  height: 100%;
  transform: scale(1);
  transition: transform ease-in-out 0.2s;

  &:hover {
    transform: scale(1.2);
  }

  path {
    fill: ${(p) => p.theme.lightFontColor};
  }
`

const TextWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

const Text = styled.div`
  font-size: 12px;
  line-height: 12px;
  color: ${(p) => p.theme.mediumFontColor};

  &:first-child {
    margin-bottom: 2px;
  }
`

const Author = ({ name, avatar, time }) => (
  <Root>
    <AvatarWrapper>
      {
        avatar
        ? <Avatar src={avatar} alt={name} />
        : <DefaultAvatar />
      }
    </AvatarWrapper>
    <TextWrapper>
      <Text>{name}</Text>
      <Text>vor {time}</Text>
    </TextWrapper>
  </Root>
)

export default Author
