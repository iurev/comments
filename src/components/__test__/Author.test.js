import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Author from '../Author'


it('with all attrs', () => {
  const tree = renderer
    .create(<Author
      avatar="https://example.com/image.png"
      name="John Doe"
      time="5 mins"
    />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('without attrs', () => {
  const tree = renderer
    .create(<Author />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})

it('without avatar', () => {
  const tree = renderer
    .create(<Author
      name="John Doe"
      time="5 mins"
    />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
