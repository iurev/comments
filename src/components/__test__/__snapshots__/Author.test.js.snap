// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with all attrs 1`] = `
<div
  className="sc-bdVaJa msRzE"
>
  <div
    className="sc-bwzfXH fqMtKx"
  >
    <img
      alt="John Doe"
      className="sc-htpNat kOUdUN"
      src="https://example.com/image.png"
    />
  </div>
  <div
    className="sc-ifAKCX dakLHt"
  >
    <div
      className="sc-EHOje bTdhSb"
    >
      John Doe
    </div>
    <div
      className="sc-EHOje bTdhSb"
    >
      vor 
      5 mins
    </div>
  </div>
</div>
`;

exports[`without attrs 1`] = `
<div
  className="sc-bdVaJa msRzE"
>
  <div
    className="sc-bwzfXH fqMtKx"
  >
    <svg
      aria-hidden="true"
      className="sc-bxivhb gXrBqi StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.579 2 2 6.579 2 12s4.579 10 10 10 10-4.579 10-10S17.421 2 12 2zm0 18c-2.151 0-4.135-.902-5.59-2.34.753-1.807 2.515-3.088 4.59-3.088h2c2.075 0 3.837 1.281 4.59 3.088C16.135 19.098 14.151 20 12 20zm0-13c1.727 0 3 1.272 3 3s-1.273 3-3 3c-1.726 0-3-1.272-3-3s1.274-3 3-3z"
      />
    </svg>
  </div>
  <div
    className="sc-ifAKCX dakLHt"
  >
    <div
      className="sc-EHOje bTdhSb"
    />
    <div
      className="sc-EHOje bTdhSb"
    >
      vor 
    </div>
  </div>
</div>
`;

exports[`without avatar 1`] = `
<div
  className="sc-bdVaJa msRzE"
>
  <div
    className="sc-bwzfXH fqMtKx"
  >
    <svg
      aria-hidden="true"
      className="sc-bxivhb gXrBqi StyledIconBase-sc-bdy9j4 jbPeIR"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.579 2 2 6.579 2 12s4.579 10 10 10 10-4.579 10-10S17.421 2 12 2zm0 18c-2.151 0-4.135-.902-5.59-2.34.753-1.807 2.515-3.088 4.59-3.088h2c2.075 0 3.837 1.281 4.59 3.088C16.135 19.098 14.151 20 12 20zm0-13c1.727 0 3 1.272 3 3s-1.273 3-3 3c-1.726 0-3-1.272-3-3s1.274-3 3-3z"
      />
    </svg>
  </div>
  <div
    className="sc-ifAKCX dakLHt"
  >
    <div
      className="sc-EHOje bTdhSb"
    >
      John Doe
    </div>
    <div
      className="sc-EHOje bTdhSb"
    >
      vor 
      5 mins
    </div>
  </div>
</div>
`;
